src/Core/Binding/IServerConnectionsRepository.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+
+namespace SonarLint.VisualStudio.Core.Binding;
+
+public interface IServerConnectionsRepository
+{
+    bool TryGet(string connectionId, out ServerConnection serverConnection);
```

Comment:
I think all methods should return Task

Reply:
I'm not sure. In ISolutionBindingRepository all operations are synchronous. It doesn't mean that we should not change that, but is there a good reason to do so? Why do you think having async here would be worth it? Anything that touches this will also have to be async due to that

Reply:
Good point about the UI. Also agree that we can change that later if needed.